src/Core/3DTiles/C3DTBatchTable.js

```diff
@@ -0,0 +1,74 @@
+/** @classdesc
+ * A 3D Tiles
+ * [batch
+ * table](https://github.com/AnalyticalGraphicsInc/3d-tiles/tree/master/specification/TileFormats/BatchTable).
+ * @property {boolean} isC3DTBatchTable - Used by 3D Tiles extensions
+ * (e.g. {@link C3DTBatchTableHierarchyExtension}) to know in which context
+ * (i.e. for which 3D Tiles class) the parsing of the extension should be done.
+ * This value is set to true by default.
+ * @property {number} batchLength - the length of the batch.
+ * @property {object} content - the content of the batch table in the form:
+ * {property1: values[], property2: values[], ...}.
+ * @property {object} extensions - Extensions of the batch table in the form:
+ * {extensioName1: extensionObject1, extensioName2: extensionObject2, ...}
+ */
+class C3DTBatchTable {
+    /**
+     * @param {Object} json - batch table json part
+     * @param {ArrayBuffer} binary - buffer representing the batch table
+     * binary part (not supported yet)
+     * @param {number} batchLength - the length of the batch.
+     * @param {Object} registeredExtensions - extensions registered to the layer
+     */
+    constructor(json, binary, batchLength, registeredExtensions) {
+        this.isC3DTBatchTable = true;
+        if (binary !== undefined) {
+            console.warn('Binary batch table content not supported yet.');
+        }
+        this.batchLength = batchLength;
+
+        // Separate the content and the possible extensions
+        // When an extension is found, we call its parser and append the
+        // returned object to batchTable.extensions
+        // Extensions must be registered in the layer (see an example of this in
+        // 3dtiles_hierarchy.html)
+        if (json.extensions) {
+            this.extensions =
+                registeredExtensions.parseExtensions(json.extensions, this);
```

Comment:
I am not sure to understand the beginning of your comment regarding adding a `type` attribute to the Extensions classes (e.g. `C3DTBatchTableHierarchyExtension`) to avoid passing the context; could you elaborate a bit please?

I agree that we should remove the `BatchTableParser` and move its content to `C3DTBatchTable`; I initially planned to do it in another PR but I will do it here.

Reply:
[see comment](https://github.com/iTowns/itowns/pull/1387#discussion_r427881753)

Reply:
`BatchTableParser` removed.